cmd/sovereignnode/chainSimulator/tests/bridge/bridge_test.go

```diff
@@ -55,12 +75,27 @@ func TestSovereignChainSimulator_DeployBridgeContractsThenIssueAndDeposit(t *tes
 
 	nodeHandler := cs.GetNodeHandler(core.SovereignChainShardId)
 
-	wallet, err := cs.GenerateAndMintWalletAddress(core.SovereignChainShardId, chainSim.InitialAmount)
+	initialAddress := "erd1l6xt0rqlyzw56a3k8xwwshq2dcjwy3q9cppucvqsmdyw8r98dz3sae0kxl"
+	initialAddrBytes, err := cs.GetNodeHandler(0).GetCoreComponents().AddressPubKeyConverter().Decode(initialAddress)
+	require.Nil(t, err)
+
+	err = cs.SetStateMultiple([]*dtos.AddressState{
+		{
+			Address: initialAddress,
+			Balance: "10000000000000000000000",
+		},
+	})
+	require.Nil(t, err)
+
+	wallet := dtos.WalletAddress{Bech32: initialAddress, Bytes: initialAddrBytes}
+
+	expectedESDTSafeAddressBytes, err := nodeHandler.GetCoreComponents().AddressPubKeyConverter().Decode(outGoingSubscribedAddress)
 	require.Nil(t, err)
-	nonce := uint64(0)
 
-	bridgeData := DeploySovereignBridgeSetup(t, cs, esdtSafeWasmPath, feeMarketWasmPath)
+	bridgeData := DeploySovereignBridgeSetup(t, cs, wallet, esdtSafeWasmPath, feeMarketWasmPath)
```

Comment:
I would just send a bech32 address as parameter (maybe a common constant in common.go) and do the SetStateMultiple inside the deploy function, and return OwnerWallet: dtos.WalletAddress as another parameter in case we want to make other transactions on the bridge contracts.

Reply:
1. It needs now bech32
2. Wallet should be input paramter, not randomly generated there and returned.

Reply:
Yes, you could send `initialAddress` as parameter and do the set state and stuff inside the `DeploySovereignBridgeSetup` and return the wallet+nonce in `bridgeData: ArgsBridgeSetup`. Then the rest of the transactions could be done from bridge owner wallet or a new random wallet.

Anyways, it's fine as it is, maybe we'll do this later.

Reply:
1. Why would you generate a random wallet, since you want to set in genesis the exact sc address that will be deployed? no need to use extra wallets.
2. No need to return the nonce, added func to retrieve the nonce